src/misc_gui.cpp

```diff
@@ -928,6 +935,13 @@ struct QueryStringWindow : public Window
 			resize.width = 0;
 			size.width = 0;
 		}
+
+		if (widget == WID_QS_MOVE && !this->flags.Test(QueryStringFlag::EnableMove)) {
+			/* We don't want this widget to show! */
+			fill.width = 0;
+			resize.width = 0;
+			size.width = 0;
+		}
```

Comment:
We normally hide widgets by wrapping them in an `NWID_SELECTION` and then using `SetDisplayedPlane()`. You can look in `rail_gui.cpp` to see how this is done for the signal menu.